portal/src/screens/api-resources/APIResourceDetailsScreenApplicationsTab.tsx

```diff
@@ -1,5 +1,57 @@
-import React from "react";
+import React, { useMemo } from "react";
+import { FormattedMessage } from "@oursky/react-messageformat";
+import WidgetTitle from "../../WidgetTitle";
+import { Text } from "@fluentui/react";
+import { Resource } from "../../graphql/adminapi/globalTypes.generated";
+import {
+  ApplicationList,
+  ApplicationListItem,
+} from "../../components/api-resources/ApplicationList";
+import { useAppAndSecretConfigQuery } from "../../graphql/portal/query/appAndSecretConfigQuery";
+import { useParams } from "react-router-dom";
+import ShowError from "../../ShowError";
 
-export function APIResourceDetailsScreenApplicationsTab(): JSX.Element {
-  return <div>TODO</div>;
+export function APIResourceDetailsScreenApplicationsTab({}: {
+  resource: Resource;
+}): JSX.Element {
+  const { appID } = useParams() as { appID: string };
+  const appConfigQuery = useAppAndSecretConfigQuery(appID);
+
+  const isLoading = appConfigQuery.isLoading;
+
+  const applications = useMemo((): ApplicationListItem[] => {
+    return (
+      appConfigQuery.effectiveAppConfig?.oauth?.clients?.map(
```

Comment:
We should only show applications whose `x_application_type=m2m`

Reply:
Should be handled in later commits. Only m2m and confidential clients are shown.